app/plugin.go

```diff
@@ -77,30 +106,103 @@ func (a *App) setPluginsActive(activate bool) {
 			continue
 		}
 
-		id := plugin.Manifest.Id
+		enabled := false
+		if state, ok := a.Config().PluginSettings.PluginStates[plugin.Manifest.Id]; ok {
+			enabled = state.Enable
+		}
 
-		pluginState := &model.PluginState{Enable: false}
-		if state, ok := a.Config().PluginSettings.PluginStates[id]; ok {
-			pluginState = state
+		result := <-a.Srv.Store.Plugin().CreatePluginStatus(&model.PluginStatus{
+			ClusterDiscoveryId: a.GetClusterId(),
+			PluginId:           plugin.Manifest.Id,
+			PluginPath:         filepath.Dir(plugin.ManifestPath),
+			IsSandboxed:        a.IsPluginSandboxSupported,
+			Name:               plugin.Manifest.Name,
+			Description:        plugin.Manifest.Description,
+			Version:            plugin.Manifest.Version,
+		})
+		if result.Err != nil {
+			mlog.Error("failed to register plugin status", mlog.String("plugin_id", plugin.Manifest.Id), mlog.Err(result.Err))
 		}
 
-		active := a.PluginEnv.IsPluginActive(id)
+		if activate && enabled {
+			a.setPluginActive(plugin, activate)
+		} else if !activate {
+			a.setPluginActive(plugin, activate)
+		}
+	}
+
+	if err := a.notifyPluginStatusesChanged(); err != nil {
+		mlog.Error("failed to notify plugin status changed", mlog.Err(err))
+	}
+}
+
+func (a *App) setPluginActiveById(id string, activate bool) {
+	plugins, err := a.PluginEnv.Plugins()
+	if err != nil {
+		mlog.Error(fmt.Sprintf("Cannot setPluginActiveById(%t)", activate), mlog.String("plugin_id", id), mlog.Err(err))
+		return
+	}
 
-		if activate && pluginState.Enable && !active {
+	for _, plugin := range plugins {
+		if plugin.Manifest != nil && plugin.Manifest.Id == id {
+			a.setPluginActive(plugin, activate)
+		}
+	}
+}
+
+func (a *App) setPluginActive(plugin *model.BundleInfo, activate bool) {
+	if plugin.Manifest == nil {
+		return
```

Comment:
Despite ostensibly improving plugin error handling in these PRs, handling `nil` Manifests is something I'd like to circle back on. Normally, this is caught on upload, but might yet still fail for a sideloaded plugin. The `PluginStatuses` table has the plugin path to be able to report on this, but I'm currently not registering these plugins at all.